lib/import/brca/providers/st_george/st_george_handler.rb

```diff
@@ -1,436 +1,348 @@
-                           (?<rndbrc>\))?(?<sqrbc>\])?/ix.freeze
-
-          DEPRECATED_BRCA_NAMES_MAP = { 'BR1'    => 'BRCA1',
-                                        'B1'     => 'BRCA1',
-                                        'BRCA 1' => 'BRCA1',
-                                        'BR2'    => 'BRCA2',
-                                        'B2'     => 'BRCA2',
-                                        'BRCA 2' => 'BRCA2' }.freeze
-
-          BRCA_GENES_REGEX = /(?<brca>BRCA1|
-                                     BRCA2|
-                                     ATM|
-                                     CHEK2|
-                                     PALB2|
-                                     MLH1|
-                                     MSH2|
-                                     MSH6|
-                                     MUTYH|
-                                     SMAD4|
-                                     NF1|
-                                     NF2|
-                                     SMARCB1|
-                                     LZTR1)/xi.freeze
-
-          EXON_VARIANT_REGEX = /(?<variant>del|dup|ins).+ex(?<on>on)?(?<s>s)?\s
-                                (?<exons>[0-9]+(?<dgs>-[0-9]+)?)|
-                              ex(?<on>on)?(?<s>s)?\s(?<exons>[0-9]+(?<dgs>-[0-9]+)?)\s
-                              (?<variant>del|dup|ins)|
-                              (?<variant>del|dup|ins)\sexon(?<s>s)?\s
-                              (?<exons>[0-9]+(?<dgs>\sto\s[0-9]+))|
-                              (?<variant>del|dup|ins)(?<s>\s)?(?<exons>[0-9]+(?<dgs>-[0-9]+)?)|
-                              ex(?<on>on)?(?<s>s)?\s(?<exons>[0-9]+(?<dgs>\sto\s[0-9]+)?)\s
-                              (?<variant>del|dup|ins)/ix.freeze
-
-          DEPRECATED_BRCA_NAMES_REGEX = /B1|BR1|BRCA\s1|B2|BR2|BRCA\s2/i.freeze
-
-          DELIMETER_REGEX = /[&\n+,;]|and|IFD/i.freeze
+          include Import::Helpers::Brca::Providers::Rj7::Constants
 
           def process_fields(record)
             genotype = Import::Brca::Core::GenotypeBrca.new(record)
-            genotype.add_passthrough_fields(record.mapped_fields,
-                                            record.raw_fields,
-                                            PASS_THROUGH_FIELDS)
-            add_organisationcode_testresult(genotype)
-            add_moleculartestingtype(genotype, record)
-            process_genetictestcope(genotype, record)
-            res = process_variants_from_record(genotype, record)
-            res.each { |cur_genotype| @persister.integrate_and_store(cur_genotype) }
-          end
-
-          def add_organisationcode_testresult(genotype)
-            genotype.attribute_map['organisationcode_testresult'] = '697N0'
-          end
+            # records using new importer should only have SRIs starting with V
+            return unless record.raw_fields['servicereportidentifier'].start_with?('V')
```

Comment:
return statement should be the first one, no need to create genotype object if we are not processing file @lauramccluskey1

Reply:
Updated in [08d5202](https://github.com/NHSDigital/data_management_system/pull/77/commits/08d5202924c702b8cacba6321dc38912a1147ecf)